spec/azure/hosted_zones.go

```diff
@@ -0,0 +1,11 @@
+package azure
+
+// HostedZones contains the Hosted Zone IDs of the cluster.
+type HostedZones struct {
+	// API is the Hosted Zone ID for the Kubernetes API.
```

Comment:
This godoc is not correct. For AWS we use IDs like `Z7ECGI0AWJZ6Q`. For Azure those are actual domain names (which is so much nicer BTW).